backend/sportsmatch/src/main/java/com/sportsmatch/services/EventService.java

```diff
@@ -96,4 +100,64 @@ public Event createEvent(EventDTO eventDTO) {
   public void deleteEventFromDatabase(Event eventById) {
     eventRepository.deleteById(eventById.getId());
   }
+
+
+  /**
+   * Retrieves the event history of the logged-in user.
+   *
+   * @param pageable contains the pagination information (page, size)
+   * @return a list of EventHistoryDTOs representing the logged-in user's event history
+   */
+  public List<EventHistoryDTO> getEventsHistory(final Pageable pageable) {
+    String loggedUserName = userService.getUserFromTheSecurityContextHolder().getName();
+
+    return eventRepository.findEventsByUser(loggedUserName, LocalDateTime.now(), pageable)
+        .stream()
+        .map(event -> eventMapper.toDTO(event, loggedUserName, checkScoreMatch(event.getPlayers())))
+        .collect(Collectors.toList());
+  }
+
+  /**
+   * Returns the checked status of the match (check the score is matching or missing)
+   *
+   * @param players who entered the event (2 playerEvent)
+   * @return the status of the match
+   * There is 4 option:
+   * -Invalid Player -> if one of the player don't present.
+   * -Waiting for ratings -> if one of the players doesn't response with the score information.
+   * -Match -> when both player submitted their result and it is match.
+   * -Mismatch -> when both players have submitted their result and it isn't a match.
+   */
+
+  public String checkScoreMatch(Set<EventPlayer> players) {
+
+    User loggedUser = userService.getUserFromTheSecurityContextHolder();
+
+    EventPlayer loggedPlayer = players.stream()
+        .filter(p -> p.getPlayer().getName().equals(loggedUser.getName()))
+        .findFirst().orElse(null);
+
+    EventPlayer otherPlayer = players.stream()
+        .filter(p -> !Objects.equals(p.getPlayer().getName(), loggedUser.getName()))
+        .findFirst()
+        .orElse(null);
+
+    if (loggedPlayer == null || otherPlayer == null) {
+      return "Invalid Player";
+    } else if (loggedPlayer.getMyScore() == null || loggedPlayer.getOpponentScore() == null ||
+        otherPlayer.getMyScore() == null || otherPlayer.getOpponentScore() == null) {
+      return "Waiting for ratings";
```

Comment:
this method is pretty complex, please add a few unit tests

Reply:
thank, i added 4 scenario.